pkg/api/defaults.go

```diff
@@ -438,16 +438,20 @@ func (p *Properties) setAgentProfileDefaults(isUpgrade, isScale bool) {
 			profile.AcceleratedNetworkingEnabledWindows = helpers.PointerToBool(DefaultAcceleratedNetworkingWindowsEnabled)
 		}
 
+		// Ensure distro is set properly for N Series SKUs no matter what, because
+		// (1) At present, "aks-docker-engine" is the only working distro base for running GPU workloads on N Series SKUs
+		// (2) Previous versions of acs-engine had working implementations using a different distro value,
+		//     so we need to hard override it in order to produce a working cluster in upgrade/scale contexts
+		if IsNSeriesSKU(p) {
+			profile.Distro = AKSDockerEngine
```

Comment:
do we currently support N series VMs in Windows agent pools? If so (or maybe it's a good idea no matter what in case that changes later) we should also check `profile.OSType != Windows`